‘nan’ is not a member of ‘std’

<!-- TEMPLATE FOR BUG REPORTS -->

### Expected behavior
Mantid should build on Ubuntu 18.04 with gcc7

### Actual behavior
It fails with error:
```
/home/nvaytet/work/mantid/temp/mantid/Framework/Algorithms/inc/MantidAlgorithms/ReflectometrySumInQ.h:61:25: error: ‘nan’ is not a member of ‘std’
     double horizon{std::nan("")};
```
### Steps to reproduce the behavior
Build mantid with gcc7

### Platforms affected
Ubuntu 18.04

--- a/Framework/Algorithms/inc/MantidAlgorithms/ReflectometrySumInQ.h
+++ b/Framework/Algorithms/inc/MantidAlgorithms/ReflectometrySumInQ.h
@@ -3,6 +3,7 @@
 
 #include "MantidAlgorithms/DllConfig.h"
 #include "MantidAPI/Algorithm.h"
+#include <cmath>
 
 namespace Mantid {
 namespace API {

